src/Propel/Generator/Reverse/PgsqlSchemaParser.php

```diff
@@ -417,7 +417,10 @@ protected function addForeignKeys(Table $table, $oid)
                 $fk = new ForeignKey($name);
                 $fk->setForeignTableCommonName($foreignTable->getCommonName());
                 if ($table->guessSchemaName() != $foreignTable->guessSchemaName()) {
-                    $fk->setForeignSchemaName($foreignTable->getSchema());
+                     if($foreignTable->getSchema())
+                            $fk->setForeignSchemaName($foreignTable->getSchema());
+                     elseif($foreignTable->getSchema() == 'public')
```

Comment:
``php
if($foreignTable->getSchema())
    $fk->setForeignSchemaName($foreignTable->getSchema());
```

When getSchema() returns a value it will be set. If it does not  return a value you check via `elseif($foreignTable->getSchema() == 'public')`, which can never be truely because you checked already in the previous if that `$foreignTable->getSchema()` returns falsely. Falsely means it can never be `== 'public'`. If you would use instead of `elseif` simple `if`, you would set it twice. Too illustrate it:

```
$schema = 'public';
if ($schema)
    $fk->setForeignSchema($schema);
if ($schema == 'public')
    $fk->setForeignSchema('public');
```

As you see, the second check is redundant.